src/Support/Types.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Spatie\Dns\Support;
+
+class Types
+{
+    public const TYPES = [
+        DNS_A => 'A',
+        DNS_AAAA => 'AAAA',
+        DNS_CNAME => 'CNAME',
+        DNS_NS => 'NS',
+        DNS_SOA => 'SOA',
+        DNS_MX => 'MX',
+        DNS_SRV => 'SRV',
+        DNS_TXT => 'TXT',
+        DNS_CAA => 'CAA',
+        // DNS_NAPTR => 'NAPTR',
+    ];
+
+    public function toNames(int $flags): array
+    {
+        $types = [];
+
+        foreach (self::TYPES as $flag => $type) {
+            if ($flags & $flag) {
```

Comment:
@gummibeer We don't usually use this operator. Could you make this code a bit more readable? Possible using a map operation?

Reply:
I don't see a way to do this without bit-operator as it's not only a single type I have to check but can be multiple.
`DNS_A | DNS_CNAME` should result in `['A', 'CNAME']`

I can add a comment that explains this line!?
The general process that happens in this method is that I put in a bit-map (integer) and get out an array of strings.
So converting the bit-based DNS flags to strings.

Reply:
I took a better look at the code and I understand it now 👍